configcatclient/configfetcher.py

```diff
@@ -68,4 +78,48 @@ def get_configuration_json(self, force_fetch=False):
         if etag:
             self._etag = etag
 
-        return FetchResponse(response)
+        fetch_response = FetchResponse(response)
+
+        # If there wasn't a config change, we return the response.
+        if not fetch_response.is_fetched():
+            return fetch_response
+
+        preferences = fetch_response.json().get(PREFERENCES, None)
+        if preferences is None:
+            return fetch_response
+
+        base_url = preferences.get(BASE_URL)
+
+        # If the base_url is the same as the last called one, just return the response.
+        if base_url is None or self._base_url == base_url:
+            return fetch_response
+
+        redirect = preferences.get(REDIRECT)
+        # If the base_url is overridden, and the redirect parameter is not 2 (force),
+        # the SDK should not redirect the calls and it just have to return the response.
+        if self._base_url_overridden and redirect != 2:
+            return fetch_response
+
+        # The next call should use the base_url provided in the config json
+        self._base_url = base_url
+
+        # If the redirect property != 0 (redirect not needed), try to download again with the new url
+        if redirect != 0:
+
+            if redirect == 1:
+                log.warning('Please check the data_governance parameter in the ConfigCatClient initialization. '
```

Comment:
Review this warning message

Reply:
Your data_governance parameter at ConfigCatClient initialization is not in sync with your preferences on the ConfigCat Dashboard: https://app.configcat.com/organization/data-governance. Only Organization Admins can set this preference.